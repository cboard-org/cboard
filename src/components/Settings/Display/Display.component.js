import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import Switch from '@material-ui/core/Switch';
import Paper from '@material-ui/core/Paper';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import Divider from '@material-ui/core/Divider';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';

import FullScreenDialog from '../../UI/FullScreenDialog';
import messages from './Display.messages';

import './Display.css';
import {
  DISPLAY_SIZE_STANDARD,
  DISPLAY_SIZE_LARGE,
  DISPLAY_SIZE_EXTRALARGE,
  LABEL_POSITION_ABOVE,
  LABEL_POSITION_BELOW,
  LABEL_POSITION_HIDDEN
} from './Display.constants';

import {
  FONTS_FAMILIES,
  DEFAULT_FONT_FAMILY
} from './../../../providers/ThemeProvider/ThemeProvider.constants';

const propTypes = {
  /**
   * Callback fired when clicking the back button
   */
  onClose: PropTypes.func,
  displaySettings: PropTypes.object.isRequired,
  updateDisplaySettings: PropTypes.func.isRequired
};

class Display extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      ...props.displaySettings
    };
  }

  toggleHideOutput = () => {
    this.setState({
      hideOutputActive: !this.state.hideOutputActive
    });
  };

  toggleIncreaseOutputButtons = () => {
    this.setState({
      increaseOutputButtons: !this.state.increaseOutputButtons
    });
  };

  toggleDarkTheme = () => {
    this.setState({
      darkThemeActive: !this.state.darkThemeActive
    });
  };

  onDisplaySettingsChange(displaySetting, event) {
    const {
      target: { value }
    } = event;
    this.setState({ [displaySetting]: value });
  }

  renderFontFamilySelect() {
    const name = 'fontFamily';
    const actualFont = FONTS_FAMILIES.filter(
      font => font.fontName === this.state[name]
    )[0];

    return (
      <FormControl>
        <Select
          aria-label={this.props.intl.formatMessage(messages.fontFamily)}
          id={name}
          name={name}
          value={actualFont?.fontName || DEFAULT_FONT_FAMILY}
          onChange={e => this.onDisplaySettingsChange(name, e)}
          style={{ fontFamily: actualFont?.fontFamily }}
        >
          {FONTS_FAMILIES.map(font => (
            <MenuItem
              key={font?.fontName}
              value={font?.fontName}
              style={{ fontFamily: font?.fontFamily }}
            >
              {font.fontName}
            </MenuItem>
          ))}
        </Select>
      </FormControl>
    );
  }

  renderSelect(name) {
    return (
      <FormControl>
        <Select
          aria-label={this.props.intl.formatMessage(messages[name])}
          id={name}
          name={name}
          value={this.state[name]}
          onChange={e => this.onDisplaySettingsChange(name, e)}
        >
          <MenuItem
            value={
              name === 'labelPosition'
                ? LABEL_POSITION_ABOVE
                : DISPLAY_SIZE_STANDARD
            }
          >
            {this.props.intl.formatMessage(
              name === 'labelPosition'
                ? messages[LABEL_POSITION_ABOVE]
                : messages[DISPLAY_SIZE_STANDARD]
            )}
          </MenuItem>
          <MenuItem
            value={
              name === 'labelPosition'
                ? LABEL_POSITION_BELOW
                : DISPLAY_SIZE_LARGE
            }
          >
            {this.props.intl.formatMessage(
              name === 'labelPosition'
                ? messages[LABEL_POSITION_BELOW]
                : messages[DISPLAY_SIZE_LARGE]
            )}
          </MenuItem>
          <MenuItem
            value={
              name === 'labelPosition'
                ? LABEL_POSITION_HIDDEN
                : DISPLAY_SIZE_EXTRALARGE
            }
          >
            {this.props.intl.formatMessage(
              name === 'labelPosition'
                ? messages[LABEL_POSITION_HIDDEN]
                : messages[DISPLAY_SIZE_EXTRALARGE]
            )}
          </MenuItem>
        </Select>
      </FormControl>
    );
  }

  onSubmit() {
    this.props.updateDisplaySettings(this.state);
  }

  render() {
    const { onClose } = this.props;
    return (
      <FullScreenDialog
        open
        title={<FormattedMessage {...messages.display} />}
        onClose={onClose}
        onSubmit={this.onSubmit.bind(this)}
      >
        <Paper className="Display">
          <List>
            <ListItem>
              <ListItemText
                className="Display__ListItemText"
                primary={<FormattedMessage {...messages.uiSize} />}
                secondary={<FormattedMessage {...messages.uiSizeSecondary} />}
              />
              <ListItemSecondaryAction className="Display__Options">
                {this.renderSelect('uiSize')}
              </ListItemSecondaryAction>
            </ListItem>
            <Divider />
            <ListItem>
              <ListItemText
                className="Display__ListItemText"
                primary={<FormattedMessage {...messages.fontFamily} />}
                secondary={
                  <FormattedMessage {...messages.fontFamilySecondary} />
                }
              />
              <ListItemSecondaryAction className="Display__Options">
                {this.renderFontFamilySelect()}
              </ListItemSecondaryAction>
            </ListItem>
            <Divider />
            <ListItem>
              <ListItemText
                className="Display__ListItemText"
                primary={<FormattedMessage {...messages.fontSize} />}
                secondary={<FormattedMessage {...messages.fontSizeSecondary} />}
              />
              <ListItemSecondaryAction className="Display__Options">
                {this.renderSelect('fontSize')}
              </ListItemSecondaryAction>
            </ListItem>
            <Divider />
            <ListItem>
              <ListItemText
                className="Display__ListItemText"
                primary={<FormattedMessage {...messages.outputHide} />}
                secondary={
                  <FormattedMessage {...messages.outputHideSecondary} />
                }
              />
              <ListItemSecondaryAction>
                <Switch
                  checked={this.state.hideOutputActive}
                  onChange={this.toggleHideOutput}
                  value="active"
                  color="secondary"
                />
              </ListItemSecondaryAction>
            </ListItem>
            <Divider />
            <ListItem>
              <ListItemText
                className="Display__ListItemText"
                primary={
                  <FormattedMessage {...messages.outputIncreaseButtons} />
                }
                secondary={
                  <FormattedMessage
                    {...messages.outputIncreaseButtonsSecondary}
                  />
                }
              />
              <ListItemSecondaryAction>
                <Switch
                  checked={this.state.increaseOutputButtons ? true : false}
                  onChange={this.toggleIncreaseOutputButtons}
                  value="active"
                  color="secondary"
                />
              </ListItemSecondaryAction>
            </ListItem>
            <Divider />
            <ListItem>
              <ListItemText
                className="Display__ListItemText"
                primary={<FormattedMessage {...messages.labelPosition} />}
                secondary={
                  <FormattedMessage {...messages.labelPositionSecondary} />
                }
              />
              <ListItemSecondaryAction className="Display__Options">
                {this.renderSelect('labelPosition')}
              </ListItemSecondaryAction>
            </ListItem>
            <Divider />
            <ListItem>
              <ListItemText
                className="Display__ListItemText"
                primary={<FormattedMessage {...messages.darkTheme} />}
                secondary={
                  <FormattedMessage {...messages.darkThemeSecondary} />
                }
              />
              <ListItemSecondaryAction className="Display__Options">
                <Switch
                  checked={this.state.darkThemeActive}
                  onChange={this.toggleDarkTheme}
                  value="active"
                  color="secondary"
                />
              </ListItemSecondaryAction>
            </ListItem>
          </List>
        </Paper>
      </FullScreenDialog>
    );
  }
}

Display.propTypes = propTypes;

export default Display;
