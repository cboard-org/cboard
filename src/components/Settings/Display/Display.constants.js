export const DISPLAY_SIZE_STANDARD = 'Standard';
export const DISPLAY_SIZE_LARGE = 'Large';
export const DISPLAY_SIZE_EXTRALARGE = 'ExtraLarge';

// Add label positioning options
export const LABEL_POSITION_ABOVE = 'Above';
export const LABEL_POSITION_BELOW = 'Below';
export const LABEL_POSITION_HIDDEN = 'Hidden';

export const LABEL_CASE_LOWER = 'lower';
export const LABEL_CASE_DEFAULT = 'default';
export const LABEL_CASE_UPPER = 'UPPER';

export const DISPLAY_SIZE_GRID_COLS = {
  [DISPLAY_SIZE_STANDARD]: {
    lg: 6,
    md: 6,
    sm: 5,
    xs: 4,
    xxs: 3
  },
  [DISPLAY_SIZE_LARGE]: {
    lg: 4,
    md: 4,
    sm: 3,
    xs: 2,
    xxs: 2
  },
  [DISPLAY_SIZE_EXTRALARGE]: {
    lg: 3,
    md: 3,
    sm: 2,
    xs: 1,
    xxs: 1
  }
};
