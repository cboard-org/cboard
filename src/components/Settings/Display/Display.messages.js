import { defineMessages } from 'react-intl';
import {
  DISPLAY_SIZE_STANDARD,
  DISPLAY_SIZE_LARGE,
  DISPLAY_SIZE_EXTRALARGE,
  LABEL_POSITION_ABOVE,
  LABEL_POSITION_BELOW,
  LABEL_POSITION_HIDDEN
} from './Display.constants';

export default defineMessages({
  display: {
    id: 'cboard.components.Settings.Display.display',
    defaultMessage: 'Display'
  },
  [DISPLAY_SIZE_STANDARD]: {
    id: 'cboard.components.Settings.Display.StandardSize',
    defaultMessage: 'Standard'
  },
  [DISPLAY_SIZE_LARGE]: {
    id: 'cboard.components.Settings.Display.LargeSize',
    defaultMessage: 'Large'
  },
  [DISPLAY_SIZE_EXTRALARGE]: {
    id: 'cboard.components.Settings.Display.ExtraLargeSize',
    defaultMessage: 'Extra Large'
  },
  [LABEL_POSITION_ABOVE]: {
    id: 'cboard.components.Settings.Display.LabelPositionAbove',
    defaultMessage: 'Above'
  },
  [LABEL_POSITION_BELOW]: {
    id: 'cboard.components.Settings.Display.LabelPositionBelow',
    defaultMessage: 'Below'
  },
  [LABEL_POSITION_HIDDEN]: {
    id: 'cboard.components.Settings.Display.LabelPositionHidden',
    defaultMessage: 'Hidden'
  },
  uiSize: {
    id: 'cboard.components.Settings.Display.uiSize',
    defaultMessage: 'UI Size'
  },
  uiSizeSecondary: {
    id: 'cboard.components.Settings.Display.uiSizeSecondary',
    defaultMessage: 'Elements size'
  },
  fontFamily: {
    id: 'cboard.components.Settings.Display.fontFamily',
    defaultMessage: 'Font family'
  },
  fontFamilySecondary: {
    id: 'cboard.components.Settings.Display.fontFamilySecondary',
    defaultMessage: 'Change the text font used in the entire application'
  },
  fontSize: {
    id: 'cboard.components.Settings.Display.fontSize',
    defaultMessage: 'Font Size'
  },
  fontSizeSecondary: {
    id: 'cboard.components.Settings.Display.fontSizeSecondary',
    defaultMessage: 'App font size'
  },
  labelPosition: {
    id: 'cboard.components.Settings.Display.labelPosition',
    defaultMessage: 'Label Position'
  },
  labelPositionSecondary: {
    id: 'cboard.components.Settings.Display.labelPositionSecondary',
    defaultMessage:
      'Whether labels on tiles should be visible, or positioned above or below'
  },
  outputHide: {
    id: 'cboard.components.Settings.Display.outputHide',
    defaultMessage: 'Hide the output bar'
  },
  outputHideSecondary: {
    id: 'cboard.components.Settings.Display.outputHideSecondary',
    defaultMessage: 'Hides the white bar on the top where you build a sentence.'
  },
  outputIncreaseButtons: {
    id: 'cboard.components.Settings.Display.outputIncreaseButtons',
    defaultMessage: 'Increase the size of action buttons on the output bar'
  },
  outputIncreaseButtonsSecondary: {
    id: 'cboard.components.Settings.Display.outputIncreaseButtonsSecondary',
    defaultMessage:
      'Increase the size of the action buttons that are on the white bar where you build a sentence.'
  },
  darkTheme: {
    id: 'cboard.components.Settings.Display.darkTheme',
    defaultMessage: 'Enable dark theme'
  },
  darkThemeSecondary: {
    id: 'cboard.components.Settings.Display.darkThemeSecondary',
    defaultMessage:
      'The theme specifies the color of the components, darkness of the surfaces, level of shadow, appropriate opacity of ink elements, etc.'
  }
});
