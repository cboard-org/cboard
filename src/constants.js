let host = window.location.host || '';
host = host.startsWith('www.') ? host.slice(4) : host;
const DEV_API_URL = process.env.REACT_APP_DEV_API_URL || null;
export const API_URL =
  DEV_API_URL || `${window.location.protocol}//api.${host}`;
