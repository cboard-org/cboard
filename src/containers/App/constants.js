export const DEFAULT_LOCALE = 'en';
