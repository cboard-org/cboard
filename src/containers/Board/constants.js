export const IMPORT_BOARDS = 'cboard/Board/IMPORT_BOARDS';
export const ADD_BOARD = 'cboard/Board/ADD_BOARD';
export const DELETE_BOARD = 'cboard/Board/DELETE_BOARD';
export const CHANGE_BOARD = 'cboard/Board/CHANGE_BOARD';
export const PREVIOUS_BOARD = 'cboard/Board/PREVIOUS_BOARD';
export const ADD_SYMBOL = 'cboard/Board/ADD_SYMBOL';
export const DELETE_SYMBOLS = 'cboard/Board/DELETE_SYMBOLS';
export const EDIT_SYMBOLS = 'cboard/Board/EDIT_SYMBOLS';

