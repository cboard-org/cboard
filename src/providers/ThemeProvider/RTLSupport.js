import React from 'react';

import { create } from 'jss';
import rtl from 'jss-rtl';
import JssProvider from 'react-jss/lib/JssProvider';
import { createGenerateClassName, jssPreset } from '@material-ui/core/styles';

// Configure JSS
const jss = create({
  plugins: [...jssPreset().plugins, rtl()]
});

// Custom Material-UI class name generator.
const generateClassName = createGenerateClassName();

export function RTLSupport({ children }) {
  return (
    <JssProvider jss={jss} generateClassName={generateClassName}>
      {React.Children.only(children)}
    </JssProvider>
  );
}

export default RTLSupport;
