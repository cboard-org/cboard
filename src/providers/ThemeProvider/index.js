export { default } from './ThemeProvider.container';
