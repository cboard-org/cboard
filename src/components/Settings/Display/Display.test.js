import React from 'react';
import { shallowMatchSnapshot } from '../../../common/test_utils';
import { DISPLAY_SIZE_STANDARD } from './Display.constants';
import Display from './Display.component';
import { DEFAULT_FONT_FAMILY } from '../../../providers/ThemeProvider/ThemeProvider.constants';

jest.mock('./Display.messages', () => {
  return {
    display: {
      id: 'cboard.components.Settings.Display.display',
      defaultMessage: 'Display'
    },
    Standard: {
      id: 'cboard.components.Settings.Display.StandardSize',
      defaultMessage: 'Standard'
    },
    Large: {
      id: 'cboard.components.Settings.Display.LargeSize',
      defaultMessage: 'Large'
    },
    Extralarge: {
      id: 'cboard.components.Settings.Display.ExtraLargeSize',
      defaultMessage: 'Extra Large'
    },
    uiSize: {
      id: 'cboard.components.Settings.Display.uiSize',
      defaultMessage: 'UI Size'
    },
    uiSizeSecondary: {
      id: 'cboard.components.Settings.Display.uiSizeSecondary',
      defaultMessage: 'Elements size'
    },
    fontSize: {
      id: 'cboard.components.Settings.Display.fontSize',
      defaultMessage: 'Font Size'
    },
    fontSizeSecondary: {
      id: 'cboard.components.Settings.Display.fontSizeSecondary',
      defaultMessage: 'App font size'
    }
  };
});

const COMPONENT_PROPS = {
  displaySettings: {
    uiSize: DISPLAY_SIZE_STANDARD,
    fontFamily: DEFAULT_FONT_FAMILY,
    fontSize: DISPLAY_SIZE_STANDARD
  },
  intl: {
    formatMessage: msg => msg
  },
  onClose: () => {},
  updateDisplaySettings: jest.fn()
};

describe('Display tests', () => {
  test('default renderer', () => {
    shallowMatchSnapshot(<Display {...COMPONENT_PROPS} />);
  });
});
