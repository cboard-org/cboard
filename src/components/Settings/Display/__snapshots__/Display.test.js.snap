// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display tests default renderer 1`] = `
<WithStyles(FullScreenDialog)
  onClose={[Function]}
  onSubmit={[Function]}
  open={true}
  title={
    <FormattedMessage
      defaultMessage="Display"
      id="cboard.components.Settings.Display.display"
      values={Object {}}
    />
  }
>
  <WithStyles(Paper)
    className="Display"
  >
    <WithStyles(List)>
      <WithStyles(ListItem)>
        <WithStyles(ListItemText)
          primary={
            <FormattedMessage
              defaultMessage="UI Size"
              id="cboard.components.Settings.Display.uiSize"
              values={Object {}}
            />
          }
          secondary={
            <FormattedMessage
              defaultMessage="Elements size"
              id="cboard.components.Settings.Display.uiSizeSecondary"
              values={Object {}}
            />
          }
        />
        <WithStyles(ListItemSecondaryAction)
          className="Display__Options"
        >
          <RadioGroup
            aria-label={
              Object {
                "defaultMessage": "UI Size",
                "id": "cboard.components.Settings.Display.uiSize",
              }
            }
            className="Display__Options__Row"
            name="uiSize"
            onChange={[Function]}
            value="Standard"
          >
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              label={
                Object {
                  "defaultMessage": "Standard",
                  "id": "cboard.components.Settings.Display.StandardSize",
                }
              }
              labelPlacement="start"
              value="Standard"
            />
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              label={
                Object {
                  "defaultMessage": "Large",
                  "id": "cboard.components.Settings.Display.LargeSize",
                }
              }
              labelPlacement="start"
              value="Large"
            />
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              labelPlacement="start"
              value="ExtraLarge"
            />
          </RadioGroup>
        </WithStyles(ListItemSecondaryAction)>
      </WithStyles(ListItem)>
      <WithStyles(ListItem)>
        <WithStyles(ListItemText)
          primary={
            <FormattedMessage
              defaultMessage="Font Size"
              id="cboard.components.Settings.Display.fontSize"
              values={Object {}}
            />
          }
          secondary={
            <FormattedMessage
              defaultMessage="App font size"
              id="cboard.components.Settings.Display.fontSizeSecondary"
              values={Object {}}
            />
          }
        />
        <WithStyles(ListItemSecondaryAction)
          className="Display__Options"
        >
          <RadioGroup
            aria-label={
              Object {
                "defaultMessage": "Font Size",
                "id": "cboard.components.Settings.Display.fontSize",
              }
            }
            className="Display__Options__Row"
            name="fontSize"
            onChange={[Function]}
            value="Standard"
          >
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              label={
                Object {
                  "defaultMessage": "Standard",
                  "id": "cboard.components.Settings.Display.StandardSize",
                }
              }
              labelPlacement="start"
              value="Standard"
            />
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              label={
                Object {
                  "defaultMessage": "Large",
                  "id": "cboard.components.Settings.Display.LargeSize",
                }
              }
              labelPlacement="start"
              value="Large"
            />
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              labelPlacement="start"
              value="ExtraLarge"
            />
          </RadioGroup>
        </WithStyles(ListItemSecondaryAction)>
      </WithStyles(ListItem)>
    </WithStyles(List)>
  </WithStyles(Paper)>
</WithStyles(FullScreenDialog)>
`;

exports[`Display tests options behavior 1`] = `
<WithStyles(FullScreenDialog)
  onClose={[Function]}
  onSubmit={[Function]}
  open={true}
  title={
    <FormattedMessage
      defaultMessage="Display"
      id="cboard.components.Settings.Display.display"
      values={Object {}}
    />
  }
>
  <WithStyles(Paper)
    className="Display"
  >
    <WithStyles(List)>
      <WithStyles(ListItem)>
        <WithStyles(ListItemText)
          primary={
            <FormattedMessage
              defaultMessage="UI Size"
              id="cboard.components.Settings.Display.uiSize"
              values={Object {}}
            />
          }
          secondary={
            <FormattedMessage
              defaultMessage="Elements size"
              id="cboard.components.Settings.Display.uiSizeSecondary"
              values={Object {}}
            />
          }
        />
        <WithStyles(ListItemSecondaryAction)
          className="Display__Options"
        >
          <RadioGroup
            aria-label={
              Object {
                "defaultMessage": "UI Size",
                "id": "cboard.components.Settings.Display.uiSize",
              }
            }
            className="Display__Options__Row"
            name="uiSize"
            onChange={[Function]}
            value="Standard"
          >
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              label={
                Object {
                  "defaultMessage": "Standard",
                  "id": "cboard.components.Settings.Display.StandardSize",
                }
              }
              labelPlacement="start"
              value="Standard"
            />
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              label={
                Object {
                  "defaultMessage": "Large",
                  "id": "cboard.components.Settings.Display.LargeSize",
                }
              }
              labelPlacement="start"
              value="Large"
            />
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              labelPlacement="start"
              value="ExtraLarge"
            />
          </RadioGroup>
        </WithStyles(ListItemSecondaryAction)>
      </WithStyles(ListItem)>
      <WithStyles(ListItem)>
        <WithStyles(ListItemText)
          primary={
            <FormattedMessage
              defaultMessage="Font Size"
              id="cboard.components.Settings.Display.fontSize"
              values={Object {}}
            />
          }
          secondary={
            <FormattedMessage
              defaultMessage="App font size"
              id="cboard.components.Settings.Display.fontSizeSecondary"
              values={Object {}}
            />
          }
        />
        <WithStyles(ListItemSecondaryAction)
          className="Display__Options"
        >
          <RadioGroup
            aria-label={
              Object {
                "defaultMessage": "Font Size",
                "id": "cboard.components.Settings.Display.fontSize",
              }
            }
            className="Display__Options__Row"
            name="fontSize"
            onChange={[Function]}
            value="Standard"
          >
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              label={
                Object {
                  "defaultMessage": "Standard",
                  "id": "cboard.components.Settings.Display.StandardSize",
                }
              }
              labelPlacement="start"
              value="Standard"
            />
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              label={
                Object {
                  "defaultMessage": "Large",
                  "id": "cboard.components.Settings.Display.LargeSize",
                }
              }
              labelPlacement="start"
              value="Large"
            />
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              labelPlacement="start"
              value="ExtraLarge"
            />
          </RadioGroup>
        </WithStyles(ListItemSecondaryAction)>
      </WithStyles(ListItem)>
    </WithStyles(List)>
  </WithStyles(Paper)>
</WithStyles(FullScreenDialog)>
`;

exports[`Display tests options behavior 2`] = `
<WithStyles(FullScreenDialog)
  onClose={[Function]}
  onSubmit={[Function]}
  open={true}
  title={
    <FormattedMessage
      defaultMessage="Display"
      id="cboard.components.Settings.Display.display"
      values={Object {}}
    />
  }
>
  <WithStyles(Paper)
    className="Display"
  >
    <WithStyles(List)>
      <WithStyles(ListItem)>
        <WithStyles(ListItemText)
          primary={
            <FormattedMessage
              defaultMessage="UI Size"
              id="cboard.components.Settings.Display.uiSize"
              values={Object {}}
            />
          }
          secondary={
            <FormattedMessage
              defaultMessage="Elements size"
              id="cboard.components.Settings.Display.uiSizeSecondary"
              values={Object {}}
            />
          }
        />
        <WithStyles(ListItemSecondaryAction)
          className="Display__Options"
        >
          <RadioGroup
            aria-label={
              Object {
                "defaultMessage": "UI Size",
                "id": "cboard.components.Settings.Display.uiSize",
              }
            }
            className="Display__Options__Row"
            name="uiSize"
            onChange={[Function]}
            value="something"
          >
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              label={
                Object {
                  "defaultMessage": "Standard",
                  "id": "cboard.components.Settings.Display.StandardSize",
                }
              }
              labelPlacement="start"
              value="Standard"
            />
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              label={
                Object {
                  "defaultMessage": "Large",
                  "id": "cboard.components.Settings.Display.LargeSize",
                }
              }
              labelPlacement="start"
              value="Large"
            />
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              labelPlacement="start"
              value="ExtraLarge"
            />
          </RadioGroup>
        </WithStyles(ListItemSecondaryAction)>
      </WithStyles(ListItem)>
      <WithStyles(ListItem)>
        <WithStyles(ListItemText)
          primary={
            <FormattedMessage
              defaultMessage="Font Size"
              id="cboard.components.Settings.Display.fontSize"
              values={Object {}}
            />
          }
          secondary={
            <FormattedMessage
              defaultMessage="App font size"
              id="cboard.components.Settings.Display.fontSizeSecondary"
              values={Object {}}
            />
          }
        />
        <WithStyles(ListItemSecondaryAction)
          className="Display__Options"
        >
          <RadioGroup
            aria-label={
              Object {
                "defaultMessage": "Font Size",
                "id": "cboard.components.Settings.Display.fontSize",
              }
            }
            className="Display__Options__Row"
            name="fontSize"
            onChange={[Function]}
            value="Standard"
          >
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              label={
                Object {
                  "defaultMessage": "Standard",
                  "id": "cboard.components.Settings.Display.StandardSize",
                }
              }
              labelPlacement="start"
              value="Standard"
            />
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              label={
                Object {
                  "defaultMessage": "Large",
                  "id": "cboard.components.Settings.Display.LargeSize",
                }
              }
              labelPlacement="start"
              value="Large"
            />
            <WithStyles(FormControlLabel)
              control={<WithStyles(Radio) />}
              labelPlacement="start"
              value="ExtraLarge"
            />
          </RadioGroup>
        </WithStyles(ListItemSecondaryAction)>
      </WithStyles(ListItem)>
    </WithStyles(List)>
  </WithStyles(Paper)>
</WithStyles(FullScreenDialog)>
`;
